File: testing/runone.py
import sys
sys.path.insert(0,'../')
from minFunc import minFunc

def runone(filename):
    with open(filename) as fid:
        line = fid.readline()
        funname = line
        line = fid.readline()
        x0 = np.fromstring(line,sep=' ')
        options = {}
        args = []
        procargs = False
        line = fid.readline()
        while line:
            if line == '':
                procargs = true
            elif procargs:
                args.append(line)
            else:
                vals = line.split()
                fname = vals[0]
                fvalstr = vals[1]
                fval = vals[1]
                for type in (int,float):
                    try:
                        fval = type(fvalstr)
                    except ValueError:
                        continue
                options[fname] = fval
            line = fid.readline()

    print(options)

    print('===START HERE')
    if len(procargs)==0:
        eval('(x,f,exitflag,output) = minFunc('+funname+',x0,options)')
    else:
        eval('(x,f,exitflag,output) = minFunc('+funname+',x0,options'+','.join(args)+')')

    print('x',x)
    print('f',f)
    print('exitflag',exitflag)
    print('output.iterations',output.iterations)
    print('output.funcCount ',output.funcCount)
    print('output.algorithm ',output.algorithm)
    print('output.firstorderopt ',output.firstorderopt)
    print('output.message ',output.message)
    print('output.trace.fval ',' '.join(output.trace.fval))
    print('output.trace.funcCount ',' '.join(output.trace.funcCount))
    print('output.trace.optCond ',' '.join(output.trace.optCond))
    print('===END HERE\n')

runone(argv[0])

File: testing/runone.m
function runone(filename)

addpath('../minFunc_2012');
addpath('../minFunc_2012/autoDif');
addpath('../minFunc_2012/minFunc');

fid = fopen(filename);
line = fgetl(fid);
funname = line;
line = fgetl(fid);
x0 = str2num(line);
options = [];
args = {};
procargs = false;
line = fgetl(fid);
while ischar(line)
	if isempty(line)
		procargs = true;
	elseif procargs
		args{end+1} = line;
	else
		vals = textscan(line,'%s %s');
		fname = vals{1}{1};
		fvalstr = vals{2}{1};
		[fval,isnum] = str2num(fvalstr);
		if isnum
			options = setfield(options,fname,fval);
		else
			options = setfield(options,fname,fvalstr);
		end
	end
	line = fgetl(fid)
end

display(options)

fprintf('===START HERE\n')
if length(procargs)==0
	eval(['[x,f,exitflag,output] = minFunc(@' funname ',x0,options);'])
else
	eval(['[x,f,exitflag,output] = minFunc(@' funname ',x0,options' strjoin(args,',') ');'])
end

fprintf('x [')
fprintf('%f ',x(1:end-1))
fprintf('%f]\n',x(end))
fprintf('f %f\n',f)
fprintf('exitflag %d\n',exitflag)
fprintf('output.iterations %d\n',output.iterations)
fprintf('output.funcCount %d\n',output.funcCount)
fprintf('output.algorithm %d\n',output.algorithm)
fprintf('output.firstorderopt %f\n',output.firstorderopt)
fprintf('output.message %s\n',output.message)
fprintf('output.trace.fval ')
fprintf('%f ',output.trace.fval)
fprintf('\n')
fprintf('output.trace.funcCount ')
fprintf('%d ',output.trace.funcCount)
fprintf('\n')
fprintf('output.trace.optCond ')
fprintf('%f ',output.trace.optCond)
fprintf('\n')
fprintf('===END HERE\n')
exit(0)










